File: sn-search-interface/src/main/java/com/sn/search/DialogueService.java
package com.sn.search;
import com.sn.entity.PageResult;
import com.sn.pojo.QaDialogue;

import java.util.List;


/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface DialogueService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<QaDialogue> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(QaDialogue dialogue);
	
	
	/**
	 * 修改
	 */
	public void update(QaDialogue dialogue);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public QaDialogue findOne(Integer id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Integer[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(QaDialogue dialogue, int pageNum, int pageSize);

	/**
	 * 更新数据
	 */
	public void uodateshuju();
	
}

File: sn-hd-controler/src/main/webapp/service/dialogueService.js
//服务层
app.service('dialogueService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../dialogue/findAll.do');		
	};
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../dialogue/findPage.do?page='+page+'&rows='+rows);
	};
	//查询实体
	this.findOne=function(id){
		return $http.get('../dialogue/findOne.do?id='+id);
	};
	//增加 
	this.add=function(entity){
		return  $http.post('../dialogue/add.do',entity );
	};
	//修改 
	this.update=function(entity){
		return  $http.post('../dialogue/update.do',entity );
	};
	//删除
	this.delete=function(ids){
		return $http.get('../dialogue/delete.do?ids='+ids);
	};
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../dialogue/search.do?page='+page+"&rows="+rows, searchEntity);
	};
	//更新solr 数据
    this.solrupdate=function(){
        return $http.post('../dialogue/updatesolr.do');
    }
});
